src/transformers/training_args.py

```diff
@@ -1615,6 +1619,39 @@ def __post_init__(self):
             if version.parse(version.parse(torch.__version__).base_version) == version.parse("2.0.0") and self.fp16:
                 raise ValueError("--optim adamw_torch_fused with --fp16 requires PyTorch>2.0")
 
+        # We need to setup the accelerator config here *before* the first call to `self.device`
```

Comment:
ah, makes sense as the `PartialState` is usually created in the first call to `self.device` itself.